goldenrun.py

```diff
@@ -418,3 +419,34 @@ def process_wildcard_faults(faultconfig, tbexec, tbinfo):
         new_fault_entry["faultlist"] = [wildcard_faults[i]]
         new_fault_entry["delete"] = False
         faultconfig.append(new_fault_entry)
+
+
+def process_single_faults(faultconfig, tbexec, tbinfo):
+    for faultentry in faultconfig:
+        for fault in faultentry["faultlist"]:
+            if not isinstance(fault.mask, dict):
+                continue
+
+            tbinfo_tb_indexed = tbinfo.set_index("id")
+            for tb in tqdm(tbexec["tb"], leave=False):
+                tb_info_asm = tbinfo_tb_indexed.at[tb, "assembler"]
+                tb_info_total_size = tbinfo_tb_indexed.at[tb, "size"]
+
+                tb_info_asm = [
+                    int(instr.split("]")[0], 16)
+                    for instr in tb_info_asm.split("[ ")[1:]
+                ]
+                tb_info_size = list(numpy.diff(tb_info_asm))
+                tb_info_size.append(
+                    tb_info_total_size - sum(tb_info_size)
+                )  # calculate the last instr size
+
+                try:
+                    fault_inx = tb_info_asm.index(fault.address)
+
+                    fault.mask = fault.mask[str(tb_info_size[fault_inx])]
+                    fault.num_bytes = tb_info_size[fault_inx]
+
+                    return
+                except ValueError:
+                    pass
```

Comment:
I think we should stop or remove the fault if we do not have a valid fault mask

Reply:
I updated this with removal of whole fault entry since removal of single fault from a fault entry would change the desired experiment.